extensions/html-language-features/server/src/modes/javascriptMode.ts

```diff
@@ -84,9 +110,11 @@ export function getJavaScriptMode(documentRegions: LanguageModelCache<HTMLDocume
 		async doValidation(document: TextDocument, settings = workspace.settings): Promise<Diagnostic[]> {
 			host.getCompilationSettings()['experimentalDecorators'] = settings && settings.javascript && settings.javascript.implicitProjectConfig.experimentalDecorators;
 			const jsDocument = jsDocuments.get(document);
-			const languageService = await host.getLanguageService(jsDocument);
-			const syntaxDiagnostics: ts.Diagnostic[] = languageService.getSyntacticDiagnostics(jsDocument.uri);
-			const semanticDiagnostics = languageService.getSemanticDiagnostics(jsDocument.uri);
+			const languageService = await host.getLanguageService(jsDocument, workspace);
+			const filePath = deschemeURI(jsDocument.uri);
+
+			const syntaxDiagnostics: ts.Diagnostic[] = languageService.getSyntacticDiagnostics(filePath);
+			const semanticDiagnostics = languageService.getSemanticDiagnostics(filePath);
 			return syntaxDiagnostics.concat(semanticDiagnostics).map((diag: ts.Diagnostic): Diagnostic => {
```

Comment:
The changes here are adding `, workspace)` and then using `filePath` whenever the LSP talks to the language service explicitly.

Reply:
This is the same pattern for the rest of the file